app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt

```diff
@@ -93,7 +94,8 @@ class BluetoothSensorManager : SensorManager {
     override val name: Int
         get() = commonR.string.sensor_name_bluetooth
     override fun getAvailableSensors(context: Context): List<SensorManager.BasicSensor> {
-        return listOf(bluetoothConnection, bluetoothState, bleTransmitter)
+        val list = listOf(bluetoothConnection, bluetoothState)
+        return if (supportsTransmitter(context)) list.plus(bleTransmitter) else list
```

Comment:
As a result of this check, the 'BLE Transmitter' sensor disappears on my device when Bluetooth is turned off. I don't think we should completely hide the sensor, perhaps a state of 'unavailable' or 'unsupported' for the sensor is better?

(also: you wouldn't be able to turn off the sensor now if it becomes unsupported)

Reply:
Hmm good point, maybe `unavailable` is the way to go then in these cases.

Reply:
So looks like this isn't a good check to add to see if the BLE transmitter is supported as it depends on BT being enabled. The current library actually catches this failure and also stops transmitting when an error is received so might be best to let the library to continue doing its thing. I will opt to add this as a attribute of the sensor to help users understand why the transmitter refuses to start because the sensor already accounts for notifying the user when BT is turned off.